freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -200,16 +205,8 @@ public Role deleteRole(String roleName) throws FreeIpaClientException {
     }
 
     public User userAdd(String user, String firstName, String lastName) throws FreeIpaClientException {
-        FreeIpaChecks.checkUserNotProtected(user, () -> String.format("User '%s' is protected and cannot be added to FreeIPA", user));
-        List<Object> flags = List.of(user);
-        Map<String, Object> params = Map.of(
-                "givenname", firstName,
-                "sn", lastName,
-                "loginshell", "/bin/bash",
-                "random", true,
-                "setattr", "krbPasswordExpiration=" + MAX_PASSWORD_EXPIRATION_DATETIME
-        );
-        return (User) invoke("user_add", flags, params, User.class).getResult();
+        return UserAddOperation.get(user, firstName, lastName).invoke(this).orElseThrow(() ->
```

Comment:
Same comment as above but this time the code looks for `FreeIpaErrorCodes.DUPLICATE_ENTRY` in the original exception.

Reply:
yes, same here, in case of useradd, duplicate_entry error should be acceptable

Reply:
same here, added it, anyway i tested batch call with userdel operation for user who is not existed in freeipa already, it seemed batch call handled it, and doesn't even thrown a notfound error